File: spring-data-requery-java/src/test/java/com/coupang/springframework/data/requery/java/domain/model/PhoneFactory.java
package com.coupang.springframework.data.requery.java.domain.model;

import java.util.function.Supplier;

/**
 * com.coupang.springframework.data.requery.java.domain.model.PhoneFactory
 *
 * @author debop
 * @since 18. 6. 4
 */
public class PhoneFactory implements Supplier<AbstractPhone> {
    @Override
    public AbstractPhone get() {
        return new Phone("");
    }
}

File: src/test/kotlin/com/coupang/springframework/data/requery/domain/functional/PhoneFactory.kt
package com.coupang.springframework.data.requery.domain.functional

import java.util.function.Supplier

/**
 * PhoneFactory
 *
 * @author debop@coupang.com
 * @since 18. 5. 31
 */
class PhoneFactory: Supplier<FuncPhone> {
    override fun get(): FuncPhone = FuncPhone().apply {
        phoneNumber = ""
        extensions = IntArray(5) { 300 + it }
    }
}